src/OptionsResolver.php

```diff
@@ -0,0 +1,401 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver as BaseOptionsResolver;
+
+/**
+ * @internal
+ */
+final class OptionsResolver extends BaseOptionsResolver
```

Comment:
If we have a interface for a fixer itself, then an interface for configuring it, and then we will have a non-interfaced final class it will break open principle here. Remember that everywhere here end-user may want to create his own Fixer (rule class). Then, he will use our DTO. Great, but he cannot modify the behaviour, if for some reason he would want to. Then, with interface, he could provide his own logic.